ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -95,10 +91,13 @@ public RequestValidator(
    *                       to support
    */
   public int validate(
+      final ServiceContext serviceContext,
       final List<ParsedStatement> statements,
       final Map<String, Object> propertyOverrides,
       final String sql
   ) {
+    requireSandbox(serviceContext);
+
     validateOverriddenConfigProperties(propertyOverrides);
     final KsqlExecutionContext ctx = requireSandbox(snapshotSupplier.get());
     final Injector injector = injectorFactory.apply(ctx, serviceContext);
```

Comment:
as part of this refactor, I think we also need to make sure that `KsqlEngine#getServiceContext` is removed - otherwise it circumvents the refactor. There is some complexity here, though, which is that when we create a sandbox we need to make sure that operations on the sandbox always use the same sandbox service context (e.g. there is a 1:1 mapping between sandbox engines and sandboxed ServiceContexts). Let me know if you want to get on a call to discuss this.

See discussion here for context: https://github.com/confluentinc/ksql/pull/2436#discussion_r265816225

Reply:
Is that because some of the sandbox clients (e.g. the topic client) accumulate state when you have multiple statements in 1 request?

Reply:
I'll throw an idea into this discussion and then run away: one option here would be to pass an interface into KsqlEngine.createSandbox that takes the current service context and returns a sandbox, that createSandbox could use internally to create the sandbox service context. This way we keep the relationship between the execution context and service context in tact.

Reply:
There is a plan to refactor the `KsqlEngine` to remove the internal `ServiceContext`. I'm not sure how that would be done yet, but I wanted to submit this partial PR due to several files I had to change. Or do you think there is a relationship on this PR?

Reply:
Based on a manual scan I think this might be safe independent of removing it from the `KsqlEngine`. For now, anywhere that is necessary that they are paired it is being accessed through `KsqlEngine#getServiceContext`.